STYLEGUIDE.md

```diff
@@ -0,0 +1,61 @@
+
+## Not Pure Markdown
+This means Docusaurus can support rendering normal markdown (*.md) files into static
+content, but it is also capable of rendering MDX/JSX into html as well. That is a powerful feature and opens the door
+to some snazzy effects, reuse of content and code, etc. This does come at a small cost of "not being pure markdown".
+
+Why is it considered a 'cost' that it's not pure markdown? That's because the markdown you look at is _supposed_ to be
+legible without processing. Sometimes though, you don't want to reinvent the wheel or copy/paste content etc. That's 
+where MDX/JSX will be amazing but that's also when you'll not be able to "just open the markdown file". So while it's
+often neat and useful, sometimes it's a drag. It also leaves all sorts of "extra" stuff at the top of the file and using
+the React components is somewhat "clunky".
+
+## Expectations for Style and Content
+
+Here is a list of expectations you should follow when writing a page backed by Docusaurus. These are in no particular
+order at this time but should be at least roughly grouped :
+
+* markdown: __DO__ wrap lines or format lines ___you are adding___ Consider wrapping at around 120 chars
+* markdown: __DO NOT__ wrap lines for content ___you are not touching___
+* markdown: __DO NOT__ format or wrap lines for an entire file automatically
+* markdown: __DO__ add [front matter](https://docusaurus.io/docs/next/markdown-features#front-matter) to every markdown
+            page. __Most importantly__, `title` and `description` but optionally consider adding and `image`
+* format: __DO__ format an entire markdown and NOTHING else (if trying to format a document)
+* format: __DO NOT__ format an entire markdown __and change__ content, format content __ONLY__ and with agreement from others
+* links: __DO__ use relative path links to the actual markdown file (../../../docs/some/05-markdown.md)
+* links: __DO NOT__ use fully qualified links pointing to the docs (https://openziti.io/...)
+* links: __DO NOT__ use the Docusaurus 'id' in a link as in: `../../../docs/some/markdown`. (Notice the lack of ".md"/".mdx")
+* images: __DO__ put images into the `/docusaurus/static/img` directory
+* images: __DO__ try to organize and name images properly
+* images: __DO__ try to use `.SVG` files as much as possible
+* images: __DO__ take care to test your image on both light and dark mode to see if it looks acceptable in both views
+* images: __DO__ try to not push too many binary (image) files over and over as it bloats the git repo
+* sidebar: __DO__ try to favor organizing pages using numerical prefixing: `01-page1.md`, `10-page2.md`
+* sidebar: __DO__ add distance between numerical prefixing to allow for easier insertion of pages (since this is a
+           pain to fix). This is done _specifically_ to allow the file system to reflect the view on site.
+* sidebar: __DO__ add a landing page to any sidebar groups via `_category_.json`. Example:

```

Comment:
This isn't always best. For example, an overview page may be desired for the landing page, but adding it as the category index will hide it from the sidebar. Alternatives include the generated index cards page.

Reply:
The generated index cards are generally useless. It's better to have a bespoke page per group, describing what the group is for etc.